particle-iot/firmware
# Escape special characters in SSIDs and passwords when sending them to ESP32

### Problem

Argon fails to connect to a WiFi network if the network's password contains special characters (`,`, `"`, `\`).

### Solution

Escape special characters in SSIDs and passwords as described in the AT command set manual (see the `+CWJAP` command).

### Steps to Test

- Change the password of your network to some string containing special characters, and make sure an Argon is able to connect to the network.

- Run [unit tests](https://github.com/particle-iot/firmware/tree/fix/escape_ssid/test/unit_tests).

### Example App

N/A

### References

- [ESP32 AT command set manual](
https://www.espressif.com/sites/default/files/documentation/ESP32_AT_Instruction_Set_and_Examples__EN.pdf)

---

### Completeness
